modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -625,6 +625,58 @@ void read_from_serial() {
     }
 }
 #endif
+
+/* Set up a timer interrupt to check lid switches' status every 20ms.
+ * Keeping the frequency high helps in getting rid of switch debounce. The timer ISR
+ * sets a flag when it's time to check the status of the switches. The loop()
+ * checks this flag and calls lid.switch_poller
+ */
+void setup_timer_interrupt()
+{
+  /* 48MHz/240 = 200kHz (8-bit divisor) | Select Generic Clock (GCLK) 4*/
+  REG_GCLK_GENDIV = GCLK_GENDIV_DIV(240) | GCLK_GENDIV_ID(4);
+  while (GCLK->STATUS.bit.SYNCBUSY);     // Wait for synchronization
+
+  REG_GCLK_GENCTRL = GCLK_GENCTRL_IDC |           // Set duty cycle to 50%
+                     GCLK_GENCTRL_GENEN |         // Enable GCLK4
+                     GCLK_GENCTRL_SRC_DFLL48M |   // Select 48MHz clock source
+                     GCLK_GENDIV_ID(4);           // Select GCLK4
+  while (GCLK->STATUS.bit.SYNCBUSY);              // Wait for synchronization
+
+  // Feed GCLK4 to TC4 & TC5
+  REG_GCLK_CLKCTRL = GCLK_CLKCTRL_CLKEN |         // Enable GCLK4 to TC4 & TC5
+                     GCLK_CLKCTRL_GEN_GCLK4 |     // Select GCLK4
+                     GCLK_CLKCTRL_ID_TC4_TC5;     // Feed the GCLK4 to TC4
+  while (GCLK->STATUS.bit.SYNCBUSY);              // Wait for synchronization
+
+  REG_TC4_CTRLA |= TC_CTRLA_MODE_COUNT8;          // Set the counter to 8-bit mode
+  while (TC4->COUNT8.STATUS.bit.SYNCBUSY);        // Wait for synchronization
+
+  REG_TC4_COUNT8_PER = 0xFA;   // Set the PER (period) reg to get 20ms interrupts (0xFA = 250)
+  while (TC4->COUNT8.STATUS.bit.SYNCBUSY);  // Wait for synchronization
+
+  NVIC_SetPriority(TC4_IRQn, 0);    // Set NVIC priority for TC4 to 0 (highest)
+  NVIC_EnableIRQ(TC4_IRQn);         // Connect TC4 to NVIC
+
+  REG_TC4_INTFLAG |= TC_INTFLAG_MC1 | TC_INTFLAG_MC0 | TC_INTFLAG_OVF;  // Clear interrupt flags
+  REG_TC4_INTENSET = TC_INTENSET_OVF;     // Enable TC4 ovf interrupt
+
+  REG_TC4_CTRLA |= TC_CTRLA_PRESCALER_DIV16 |   // Set prescaler to 16, f=12.5kHz (legal values 1,2,4,8,16,64,256,1024)
+                   TC_CTRLA_ENABLE;             // Enable TC4
+  while (TC4->COUNT8.STATUS.bit.SYNCBUSY);      // Wait for synchronization
+}
+
+volatile bool check_lid = false;
```

Comment:
This should be declared and defined at the top of the file so it doesn't get lost in the shuffle and there's no declaration-position issues